package/MDAnalysis/coordinates/GRO.py

```diff
@@ -290,17 +314,17 @@ def write(self, selection, frame=None):
         # Check for topology information
         missing_topology = []
         try:
-            names = atoms.names
+            names = selection.names
         except (AttributeError, NoDataError):
             names = itertools.cycle(('X',))
             missing_topology.append('names')
         try:
-            resnames = atoms.resnames
+            resnames = selection.resnames
```

Comment:
@kain88-de What should I name it, currently selection can be both `AtomGroup` or a `Timestep`